src/funcs.c

```diff
@@ -57,6 +57,36 @@
 
 #include <src/hookintrprtr.h>
 
+static ModuleStateHandle StateHandle;
+typedef struct {
+    Int RecursionDepth;
+    Obj ExecState;
+} ModuleState;
+
+void IncRecursionDepth(void)
+{
+    MODULE_STATE(RecursionDepth)++;
+}
+
+void DecRecursionDepth(void)
+{
+    MODULE_STATE(RecursionDepth)--;
+    /* Apparently this is wrong
+    GAP_ASSERT(MODULE_STATE(RecursionDepth) >= 0);
+    */
```

Comment:
That sounds like a bug (in the `master` branch, I mean, not in this PR).

Reply:
There is a comment in `funcs.c` line 799 that says "in interactive work the RecursionDepth could become slightly negative". I have not yet looked into figuring out why.